designs/2020-rule-tester-only/README.md

```diff
@@ -0,0 +1,174 @@
+
+Deleted or commented-out tests are hard to miss in diffs.
+An extra `only: true` could more easily sneak through, accidentally disabling all other tests.
+Calling Mocha with [`--forbid-only`](https://mochajs.org/#-forbid-only) will prevent that on CI runs, but it still requires third-party developers to opt in.
+
+## Backwards Compatibility Analysis
+
+<!--
+    How does this change affect existing ESLint users? Will any behavior
+    change for them? If so, how are you going to minimize the disruption
+    to existing users?
+-->
+
+This change is backwards compatible.
+`RuleTester` currently fails any tests that include unrecognized properties like `only`, so we have no risk of breaking existing tests.
+
+If someone has customized `RuleTester` with a custom `it()` method, we cannot assume that `it.only()` has the same semantics as Mocha.
+Their existing tests will still work, but if they wish to use this new `only` property, they would need to provide a custom `itOnly()` alongside `it()`.
+
+## Alternatives
+
+<!--
+    What other designs did you consider? Why did you decide against those?
+
+    This section should also include prior art, such as whether similar
+    projects have already implemented a similar feature.
+-->
+
+This builds upon Mocha's established `it.only()` API as prior art.
+
+The status quo has two alternatives:
+
+1. We can temporarily delete or comment out the other tests. This is tedious.
+1. We can filter through noise by scrolling through `console.log()` output or using conditional breakpoints, though the latter is not always possible.
+
+## Open Questions
+
+<!--
+    This section is optional, but is suggested for a first draft.
+
+    What parts of this proposal are you unclear about? What do you
+    need to know before you can finalize this RFC?
+
+    List the questions that you'd like reviewers to focus on. When
+    you've received the answers and updated the design to reflect them, 
+    you can remove this section.
+-->
+
+1. We could define a static `only()` convenience method on `RuleTester`.
+Given, for example, a valid string test `"var foo = 42;"`, it is easier to convert to `RuleTester.only("var foo = 42;")`, which is equivalent to `{ code: "var foo = 42;", only: true }`.
+The helper sets `only: true` on a test case argument after converting string cases to objects if necessary.
+For invalid cases that are already objects, adding `only: true` is likely easier than using the helper.
+1. ~~Should using `only` cause the process to exit `1` even if tests pass as described in "Drawbacks"?~~
+Our Makefile can call Mocha with [`--forbid-only`](https://mochajs.org/#-forbid-only) instead.
+3. In the `RuleTester.itOnly` `get` accessor, if `RuleTester[IT_ONLY]` is not customized and the global `it.only` is not a function, is throwing an error the right choice?
+We could instead pass through to `RuleTester.it`, ignoring `only` if it's not supported.
+1. Do we need to support `skip`, the inverse of `only`?
```

Comment:
Maybe we could evaluate this separately, after getting some feedback about `only`?

Reply:
Agree. I don’t think there’s a big need right now so let’s not hold up this RFC for it.

Reply:
076a9a5 answers this open question to reflect agreement that we can ship `only` by itself.